service/history/workflow/workflow_task_state_machine.go

```diff
@@ -562,6 +556,7 @@ func (m *workflowTaskStateMachine) FailWorkflowTask(
 	incrementAttempt bool,
 ) {
 	// clear stickiness whenever workflow task fails
+	incrementAttempt = incrementAttempt && !m.ms.IsStickyTaskQueueEnabled()
```

Comment:
why this? we should increase attempt regardless sticky or not.

Reply:
sticky workflow task should not be retied to the same task queue
meaning sticky workflow task should only have attempt == 1, plus the logic is based on number of attempts to decide whether to generate new events, etc

mid / long term story: we should just throw away the existing impl and start from stretch